test/infrastructure/docker/internal/docker/machine.go

```diff
@@ -399,7 +399,7 @@ func (m *Machine) SetNodeProviderID(ctx context.Context) error {
 	patch := fmt.Sprintf(`{"spec": {"providerID": %q}}`, m.ProviderID())
 	cmd := kubectlNode.Commander.Command(
 		"kubectl",
-		"--kubeconfig", "/etc/kubernetes/admin.conf",
+		"--kubeconfig", "/etc/kubernetes/kubelet.conf",
```

Comment:
Does the kubelet.conf on any machine has the rights to patch any other node?

Reply:
I wonder why we are doing this via kubectl. Is there something which prevents us from simply using a Client configured against the workload cluster API server in the controller? 
(maybe it doesn't work for the first control plane node and we get stuck?)

(cc @fabriziopandini)

Reply:
Any `kubelet.conf` seems not work, I think the kubelet.conf of a node is only able to patch himself:

Trying to patch itself: allowed (still throws an error due to invalid patch, could be ignored in this case) 
```
root@capi-quickstart-md-0-bj6ch-79956b7999-dvpx9:/# kubectl --kubeconfig /etc/kubernetes/kubelet.conf patch no capi-quickstart-md-0-bj6ch-79956b7999-dvpx9 -p '{"spec": {"providerID": "docker:////capi-quickstart-md-0-bj6ch-79956b7999-dvpx9-foo"}}'
The Node "capi-quickstart-md-0-bj6ch-79956b7999-dvpx9" is invalid: spec.providerID: Forbidden: node updates may not change providerID except from "" to valid
```

Trying to patch other node: fails because forbidden
```
root@capi-quickstart-md-0-bj6ch-79956b7999-dvpx9:/# kubectl --kubeconfig /etc/kubernetes/kubelet.conf patch no capi-quickstart-md-0-bj6ch-79956b7999-7hpkk -p '{"spec": {"providerID": "docker:////capi-quickstart-md-0-bj6ch-79956b7999-7hpkk-foo"}}'
Error from server (Forbidden): nodes "capi-quickstart-md-0-bj6ch-79956b7999-7hpkk" is forbidden: node "capi-quickstart-md-0-bj6ch-79956b7999-dvpx9" is not allowed to modify node "capi-quickstart-md-0-bj6ch-79956b7999-7hpkk"
```

This is due to the `Node` authorization mode in `kube-apiserver`: `--authorization-mode=Node,RBAC`

Reply:
Yup. That's what I expected

Reply:
I changed the code to launch the patch command from the node itself. CI is green now.

However, I don't see a real benefit here on launching the command from the node, besides reducing the number of lines of code. If we all agree on using the `controller-runtime` client, I can rework the PR.

Reply:
As I'm not really sure about why we use kubectl I would say let's just keep it that way (at least for this PR) as the current PR will definitely already fix your issue and is an improvement